test/suite/test_compress03.py

```diff
@@ -0,0 +1,113 @@
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import wttest
+from wiredtiger import stat
+from wtdataset import SimpleDataSet
+from wtscenario import make_scenarios
+
+# test_compress03.py
+#   Verify the statistics related to compression are updated as expected.
+#
+class test_compress03(wttest.WiredTigerTestCase):
+
+    compressors = [
+        ('lz4', dict(compressors='lz4')),
+        ('snappy', dict(compressors='snappy')),
+        ('zlib', dict(compressors='zlib')),
+        ('zstd', dict(compressors='zstd')),
+    ]
+    scenarios = make_scenarios(compressors)
+
+    uri = "table:test_compress03"
+    nrows = 100000
+    valuea = "aaaaa"
+    valueb = "bbbbb"
+
+    def conn_config(self):
+        return f'statistics=(fast),statistics_log=(json,on_close,wait=1,sources=(\"file:\"))'
+
+    # Load the compression extension, skip the test if missing.
+    def conn_extensions(self, extlist):
+        extlist.skip_if_missing = True
+        extlist.extension('compressors', self.compressors)
+
+    def verify_stats(self, uri):
+        stat_cursor = self.session.open_cursor('statistics:' + uri, None, 'statistics=(fast)')
+        total_reads = stat_cursor[stat.dsrc.compress_read][2]
+        total_reads_hist = stat_cursor[stat.dsrc.compress_read_hist_ratio_2][2]
+        total_reads_hist += stat_cursor[stat.dsrc.compress_read_hist_ratio_4][2]
+        total_reads_hist += stat_cursor[stat.dsrc.compress_read_hist_ratio_8][2]
+        total_reads_hist += stat_cursor[stat.dsrc.compress_read_hist_ratio_16][2]
+        total_reads_hist += stat_cursor[stat.dsrc.compress_read_hist_ratio_32][2]
+        total_reads_hist += stat_cursor[stat.dsrc.compress_read_hist_ratio_64][2]
+        total_reads_hist += stat_cursor[stat.dsrc.compress_read_hist_ratio_max][2]
+        # We expect at least a read of compressed data.
+        assert total_reads and total_reads == total_reads_hist
```

Comment:
Does it matter if all of the stats are 0? It would be nice to make sure that they aren't all 0 all the time (for these and write stats).

I suspect the read ones are commonly zero for this test.

Reply:
This is where I am struggling, I am not able to update all the stats with the current test. I need to find a pattern that will  change the compression ratio in a way that each stat is updated.